packages/paypal-express-integration/src/paypal-express-button-strategy.ts

```diff
@@ -0,0 +1,182 @@
+    PaypalStyleOptions,
+} from './paypal-express-types';
+
+export default class PaypalButtonStrategy implements CheckoutButtonStrategy {
+    constructor(
+        private paymentIntegrationService: PaymentIntegrationService,
+        private paypalExpressScriptLoader: PaypalScriptLoader,
+        private formPoster: FormPoster,
+    ) {}
+
+    async initialize(
+        options: CheckoutButtonInitializeOptions & WithPaypalExpressButtonInitializeOptions,
+    ): Promise<void> {
+        const state = this.paymentIntegrationService.getState();
+        const paymentMethod = state.getPaymentMethodOrThrow(options.methodId);
+
+        if (!options.paypal) {
+            throw new InvalidArgumentError();
+        }
+
+        if (!paymentMethod.config.merchantId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        const paypalSDK = await this.paypalExpressScriptLoader.loadPaypalSDK(
+            paymentMethod.config.merchantId,
+        );
+
+        this.renderButton(options, paypalSDK);
+    }
+
+    deinitialize(): Promise<void> {
+        return Promise.resolve();
+    }
+
+    private renderButton(
+        options: CheckoutButtonInitializeOptions & WithPaypalExpressButtonInitializeOptions,
+        paypalSDK: PaypalSDK,
+    ) {
+        const state = this.paymentIntegrationService.getState();
+        const paymentMethod = state.getPaymentMethodOrThrow(options.methodId);
+
+        if (!paymentMethod.config.merchantId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        const paypalOptions = options.paypal;
+        const merchantId = paymentMethod.config.merchantId;
+        const env = paymentMethod.config.testMode ? 'sandbox' : 'production';
+        const clientToken: PaypalClientToken = { [env]: paypalOptions.clientId };
+
+        const fundingCreditOption = paypalSDK.FUNDING.CREDIT || 'credit';
+        const allowedSources = paypalOptions.allowCredit ? [fundingCreditOption] : [];
+        const disallowedSources = !paypalOptions.allowCredit ? [fundingCreditOption] : [];
+
+        return paypalSDK.Button.render(
+            {
+                env,
+                client: clientToken,
+                commit: paypalOptions.shouldProcessPayment,
+                funding: {
+                    allowed: allowedSources,
+                    disallowed: disallowedSources,
+                },
+                style: this.getStyle(paypalOptions.style),
+                payment: (_, actions) =>
+                    this.setupPayment(merchantId, actions, paypalOptions.onPaymentError),
+                onAuthorize: (data, actions) =>
+                    this.tokenizePayment(
+                        data,
+                        options.methodId,
+                        actions,
+                        paypalOptions.shouldProcessPayment,
+                    ),
+            },
+            options.containerId,
+        );
```

Comment:
[optional] I think we can make this method a bit better by making destructurization for some variables:

```suggestion
    ) {
       const { containerId, methodId, paypal } = options;
       const {
           allowCredit,
           clientId,
           onPaymentError,
           shouldProcessPayment,
           style,
       } = paypal;
    
        const state = this.paymentIntegrationService.getState();
        const paymentMethod = state.getPaymentMethodOrThrow(options.methodId);

        if (!paymentMethod.config.merchantId) {
            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
        }

        const merchantId = paymentMethod.config.merchantId;
        const env = paymentMethod.config.testMode ? 'sandbox' : 'production';
        const clientToken: PaypalClientToken = { [env]: clientId };

        const fundingCreditOption = paypalSDK.FUNDING.CREDIT || 'credit';
        const allowedSources = allowCredit ? [fundingCreditOption] : [];
        const disallowedSources = !allowCredit ? [fundingCreditOption] : [];

        return paypalSDK.Button.render(
            {
                env,
                client: clientToken,
                commit: shouldProcessPayment,
                funding: {
                    allowed: allowedSources,
                    disallowed: disallowedSources,
                },
                style: this.getStyle(style),
                payment: (_, actions) =>
                    this.setupPayment(merchantId, actions, onPaymentError),
                onAuthorize: (data, actions) =>
                    this.tokenizePayment(
                        data,
                        methodId,
                        actions,
                        shouldProcessPayment,
                    ),
            },
            containerId,
        );
```

Reply:
fixed